src/DataProcessor.ts

```diff
@@ -92,47 +99,60 @@ export class DataProcessor implements IDataProcessor {
     // Makes the first wsman call
     const clientObj = this.clientManager.getClientObject(clientId)
     if ((clientObj.action === ClientAction.ADMINCTLMODE || clientObj.action === ClientAction.CLIENTCTLMODE) && !clientMsg.payload.digestRealm && !clientObj.activationStatus.missingMebxPassword) {
-      await this.amtwsman.batchEnum(clientId, '*AMT_GeneralSettings')
+      const messageId = (this.httpHandler.messageId++).toString()
+      const xmlRequestBody = this.amt.GeneralSettings(AMT.Methods.GET, messageId)
+      this.httpHandler.connectionParams = {
+        port: 16992,
+        guid: clientObj.ClientData.payload.uuid,
+        username: clientObj.ClientData.payload.username,
+        password: clientObj.ClientData.payload.password
+      }
+      const data = this.httpHandler.wrapIt(xmlRequestBody)
+      return this.responseMsg.get(clientId, data, 'wsman', 'ok', 'alls good!')
     } else {
       const response = await this.clientActions.buildResponseMessage(clientMsg, clientId)
       return response
     }
   }
 
-  async deactivateDevice (clientMsg: ClientMsg, clientId: string): Promise<void> {
+  async deactivateDevice (clientMsg: ClientMsg, clientId: string): Promise<ClientMsg> {
     this.logger.debug(`ProcessData: Parsed DEACTIVATION Message received from device ${clientMsg.payload.uuid}: ${JSON.stringify(clientMsg, null, '\t')}`)
     await this.validator.validateDeactivationMsg(clientMsg, clientId) // Validate the deactivation message payload
-    await this.amtwsman.getWSManResponse(clientId, 'AMT_SetupAndConfigurationService', 'admin')
+    const clientObj = this.clientManager.getClientObject(clientId)
```

Comment:
Why do we need client object now but didnt need it before?

Reply:
Resolved in deactivator PR